lint/linter.py

```diff
@@ -895,7 +909,7 @@ def get_working_dir(self, settings):
 
     def get_environment(self, settings):
         """Return runtime environment for this lint."""
-        return ChainMap({}, settings.get('env', {}), self.env)
+        return ChainMap({}, settings.get('env', {}), self.env, BASE_LINT_ENVIRONMENT)
```

Comment:
This is also nice to read. But note, that if you subclass, and use super and augment that, you can easily get the wrong shadowing.